breathecode/activity/tests/urls/tests_cohort.py

```diff
@@ -0,0 +1,295 @@
+
+#     return MagicMock(side_effect=fetch)
+
+
+def datastore_fetch_mock(first_fetch=[]):
+    class Vars():
+        fetch_call_counter = 0
+        fetch_call_one = first_fetch
+
+    Vars.fetch_call_counter = 0
+
+    def fetch(**kwargs):
+        Vars.fetch_call_counter += 1
+
+        if Vars.fetch_call_counter % 2 == 1:
+            result = Vars.fetch_call_one
+            offset = 0
+            try:
+                limit = kwargs['limit']
+                offset = kwargs['offset']
+            except:
+                return result
+            if limit is not None:
+                return result[offset:offset + limit]
+        return []
+
+    return MagicMock(side_effect=fetch)
+
+
+def datastore_count_mock(how_many):
+    def count(**kwargs):
+        return how_many
+
+    return MagicMock(side_effect=count)
+
+
+class MediaTestSuite(MediaTestCase):
+    """
+    🔽🔽🔽 With data
+    """
+    @patch.object(Datastore, '__init__', new=lambda x: None)
+    @patch.object(Datastore,
+                  'fetch',
+                  new=datastore_fetch_mock(first_fetch=DATASTORE_SEED))
+    @patch.object(Datastore, 'count', new=datastore_count_mock(7957599))
+    def test_get_activities_slug_filtered(self):
+        from breathecode.services.google_cloud import Datastore as mock
+        mock.fetch.call_args_list = []
+
+        self.headers(academy=1)
+        cohort_kwargs = {'slug': 'miami-downtown-pt-xx'}
+        self.generate_models(authenticate=True,
+                             profile_academy=True,
+                             capability='classroom_activity',
+                             role='potato',
+                             cohort_kwargs=cohort_kwargs)
+
+        url = reverse_lazy('activity:academy_cohort_id',
+                           kwargs={'cohort_id': 1}) + '?slug=breathecode_login'
+        response = self.client.get(url)
+
+        json = response.json()
+        expected = [
+            {
+                'academy_id': 0,
+                'cohort': None,
+                'created_at': DATASTORE_SEED[0]['created_at'],
+                'data': None,
+                'day': 13,
+                'email': 'konan@naruto.io',
+                'slug': 'breathecode_login',
+                'user_agent': 'bc/test',
+                'user_id': 1,
+            },
+        ]
+
+        self.assertEqual(json, expected)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(mock.fetch.call_args_list, [
+            call(kind='student_activity',
+                 cohort='miami-downtown-pt-xx',
+                 slug='breathecode_login'),
+        ])
+
+    """
+    🔽🔽🔽 Without pagination
+    """
+
+    @patch.object(Datastore, '__init__', new=lambda x: None)
+    @patch.object(Datastore,
+                  'fetch',
+                  new=datastore_fetch_mock(first_fetch=generate_data(100)))
+    @patch.object(Datastore, 'count', new=datastore_count_mock(7957599))
```

Comment:
You should prevent call `.count` and using this mock if `limit` or `offset` is not set